src/rag/embeddings.py

```
@@ -1,58 +1,132 @@
+
 
 logger = get_logger(__name__)
 
+
+
+
+# Initialize client once at module level (reads API key from environment)
+_client = genai.Client()
+
 @retry(stop=stop_after_attempt(3), wait=wait_exponential(multiplier=1, min=4, max=10))
 def get_embedding(text: str, task_type: str = "RETRIEVAL_DOCUMENT") -> Optional[List[float]]:
```

Ruff (0.13.1)

Unused function argument: `task_type`

(ARG001)